chain-impl-mockchain/tests/ledger/ledger_tests.rs

```diff
@@ -35,198 +31,200 @@ macro_rules! assert_err {
-    let (_sk2, _pk2, user2_address) =
-        accounts::make_utxo_delegation_key(&mut rng, &mut delegation_rng, &discrimination);
+    let faucet = AddressData::account(Discrimination::Test);
+    let receiver = AddressData::delegation(Discrimination::Test);
 
-    let (message, _) = ledger::create_initial_transaction(Output {
-        address: user1_address.clone(),
-        value: Value(42000),
-    });
+    let (message, _) = ledger::create_initial_transaction(Output::from_address_and_value(
+        faucet.address.clone(),
+        Value(42000),
+    ));
 
     let (block0_hash, ledger) =
         ledger::create_initial_fake_ledger(&[message], ConfigBuilder::new().build());
     let signed_tx = TransactionBuilder::new()
-        .with_input(Input::from_account(
-            AccountIdentifier::from_single_account(Identifier::from(pk1)),
+        .with_input(Input::from_account_pk(faucet.public_key.clone(), Value(1)))
+        .with_output(Output::from_address_and_value(
+            receiver.address.clone(),
             Value(1),
         ))
-        .with_output(Output {
-            address: user2_address.clone(),
-            value: Value(1),
-        })
         .finalize()
-        .with_account_witness(&block0_hash, &SpendingCounter::zero(), &sk1)
+        .with_witness(&block0_hash, &faucet)
         .seal();
 
-    let dyn_params = ledger.get_ledger_parameters();
-    let r = ledger.apply_transaction(&signed_tx, &dyn_params);
-    assert!(r.is_ok());
+    let fees = ledger.get_ledger_parameters();
+    assert!(ledger.apply_transaction(&signed_tx, &fees).is_ok());
 }
 
 #[test]
 pub fn delegation_to_account_correct_transaction() {
-    let discrimination = Discrimination::Test;
+    let faucet = AddressData::delegation(Discrimination::Test);
+    let receiver = AddressData::account(Discrimination::Test);
 
-    let mut rng = rand::thread_rng();
-    let mut delegation_rng = rand::thread_rng();
-    let (sk1, _pk1, user1_address) =
-        accounts::make_utxo_delegation_key(&mut rng, &mut delegation_rng, &discrimination);
-    let (_sk2, _pk2, user2_address) = accounts::make_account_key(&mut rng, &discrimination);
+    let (message, utxos) = ledger::create_initial_transaction(Output::from_address_and_value(
+        faucet.address.clone(),
+        Value(42000),
+    ));
 
-    let (message, utxos) = ledger::create_initial_transaction(Output {
-        address: user1_address.clone(),
-        value: Value(42000),
-    });
+    let (block0_hash, ledger) =
+        ledger::create_initial_fake_ledger(&[message], ConfigBuilder::new().build());
+
+    let signed_tx = TransactionBuilder::new()
+        .with_input(Input::from_utxo(utxos[0]))
+        .with_output(Output::from_address_and_value(
+            receiver.address.clone(),
+            Value(42000),
+        ))
+        .finalize()
+        .with_witness(&block0_hash, &faucet)
+        .seal();
```

Comment:
This would look more logical if the `finalize` method was renamed to reflect the intermediate building phase, like:

```rust
    let signed_tx = TransactionBuilder::new()
        .with_input(Input::from_utxo(utxos[0]))
        .with_output(Output::from_address_and_value(
            receiver.address.clone(),
            Value(42000),
        ))
        .authenticate()
        .with_witness(&block0_hash, &faucet)
        .seal();
```

Reply:
(Also, I was wrong and the intermediate returned object can be actually mutably borrowed)